packages/vscode-extension/src/treeview/account/azureNode.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as path from "path";
+import * as vscode from "vscode";
+
+import { SubscriptionInfo } from "@microsoft/teamsfx-api";
+
+import AzureAccountManager from "../../commonlib/azureLogin";
+import { ext } from "../../extensionVariables";
+import { TelemetryTriggerFrom } from "../../telemetry/extTelemetryEvents";
+import { localize } from "../../utils/localizeUtils";
+import { DynamicNode } from "../dynamicNode";
+import { AccountItemStatus, loadingIcon } from "./common";
+import { SubscriptionNode } from "./subscriptionNode";
+
+export class AzureAccountNode extends DynamicNode {
+  public status: AccountItemStatus;
+  private subscriptionNode: SubscriptionNode;
+
+  constructor(private eventEmitter: vscode.EventEmitter<DynamicNode | undefined | void>) {
+    super("", vscode.TreeItemCollapsibleState.None);
+    this.status = AccountItemStatus.SignedOut;
+    this.contextValue = "signinAzure";
+    this.subscriptionNode = new SubscriptionNode(this.eventEmitter);
+  }
+
+  public async setSignedIn(upn: string) {
+    if (this.status === AccountItemStatus.SignedIn) {
+      return false;
+    }
+    this.status = AccountItemStatus.SignedIn;
+    this.label = upn;
+    this.contextValue = "signedinAzure";
+    const needManualSelection = await this.autoSelectSubscription();
+    // refresh
+    this.eventEmitter.fire(this);
+    return needManualSelection;
+  }
+
+  public setSigningIn() {
+    if (this.status === AccountItemStatus.SigningIn) {
+      return;
+    }
+    this.status = AccountItemStatus.SigningIn;
+    this.contextValue = "";
+    // refresh
+    this.eventEmitter.fire(this);
+  }
+
+  public setSignedOut() {
```

Comment:
Will this remove the subscription node from Azure node?

Reply:
Yes, since the status is set to `SignedOut` here, `getTreeItem()` triggered by `eventEmitter` will update the collapsible state to `None` which removes the subscription node.